ui/src/pages/report/Report.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import Table from "@material-ui/core/Table";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import TableBody from "@material-ui/core/TableBody";
+import Button from "@material-ui/core/Button";
+import { useStreamQuery, useLedger, useParty } from "@daml/react";
+import { CreateEvent } from "@daml/ledger";
+import { Asset } from "@daml.js/daml-ui-template-0.0.1/lib/Main";
+import { InputDialog, InputDialogProps, Field } from "./InputDialog";
+import useStyles from "./styles";
+
+export default function Report() {
+  const classes = useStyles();
+
+  const [ props, setProps ] = useState<InputDialogProps>({ open: false, title: "", fields: [], onClose: async () => {} });
+  const party = useParty();
+  const ledger = useLedger();
+  const assets = useStreamQuery(Asset).contracts;
+  
+  const showGive = (asset : CreateEvent<Asset>) => {
+    const fields : Field[] = [
+      { label: "New Owner", name: "newOwner", type: { items: [ "Alice", "Bob" ] } }
+    ]
+    const onClose = async (state : any) => {
+      console.log(state);
+      setProps({ open: false, title: "", fields: [], onClose: async () => {} });
```

Comment:
I wish that we didn't have to repeat ourselves here. Also we could have a state for Give and Appraise?